ethcore/src/executive.rs

```diff
@@ -253,13 +259,17 @@ impl<'a, B: 'a + StateBackend> Executive<'a, B> {
 		// backup used in case of running out of gas
 		self.state.checkpoint();
 
+		trace!("Executive::call(params={:?}) self.env_info={:?}", params, self.info);
+		if (params.call_type == CallType::StaticCall || self.static_flag) && params.value.value() > 0.into() {
+			return Err(evm::Error::MutableCallInStaticContext);
```

Comment:
Two issues:
1. Shouldn't such calls be traced anyway?
2. Seems that `state.checkpoint` is being made, but since we are not `enacting_result` it will not be reverted, is that correct?